pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSink.java

```diff
@@ -156,36 +158,33 @@ ElasticSearchClient getElasticsearchClient() {
             }
 
             String id = null;
-            if (key != null) {
-                if (keySchema != null) {
-                    id = stringifyKey(keySchema, key);
-                } else {
-                    id = key.toString();
-                }
+            if (elasticSearchConfig.isKeyIgnore() == false && key != null && keySchema != null) {
+                id = stringifyKey(keySchema, key);
             }
 
             String doc = null;
             if (value != null) {
                 if (valueSchema != null) {
                     doc = stringifyValue(valueSchema, value);
                 } else {
-                    doc = value.toString();
+                    if (value instanceof byte[]) {
+                        // for BWC with the ES-Sink
+                        doc = new String( (byte[]) value);
```

Comment:
This is not good because value is a GenericObject, not byte[]. So we can fix this to be BWC without adding the schemaAware config switch. See the ElasticSearchBWCTests.